### ISSUE TITLE
Doxygen cannot handle `#define` after PlantUML

### ISSUE THREAD

**Issue by Yoshi-Nakai** [doxygen/doxygen#10353]
Describes bug information.

I included this as additional information to Doxygen cannot combine documents #10305,
Since I can't get any comments, I'll start a new thread.

1. Add LOCAL_MACRO to "test.c". ... ex org

The document will be generated without any problems.

2. Add her activity diagram using PlantUML to the prototype declaration of "test.h". ... ex ng1

To improve readability, add an activity diagram using PlantUML.
Since the documentation is written in the prototype declaration in "test.h",
If you add an activity diagram to it,
"warning: documentation for unknown define LOCAL_MACRO found. "
and no documentation for LOCAL_MACRO is generated.

3. Add an activity diagram to the function body of "test.c" using PlantUML. ... ex ng2

If you move only the activity diagram to the function body of "test.c", the warning will disappear, but
The html documentation for both "test.c" and "test.h" functions is different.

4. Add LOCAL_MACRO to "test.h" (before activity diagram). ... ex ok

If you move LOCAL_MACRO before the activity diagram on the "test.h" side,
A LOCAL_MACRO document will be generated on the "test.h" side.
Other than that, everything seems fine.
However, it is undesirable to expose local definitions globally in "test.h".

If you move LOCAL_MACRO after the activity diagram on the "test.h" side,
"warning: documentation for unknown define LOCAL_MACRO found. "
and no documentation for LOCAL_MACRO is generated.

Attachments.
"ex org" : Add LOCAL_MACRO to "test.c"
"ex ng1": Added an activity diagram to the prototype declaration of "test.h" in "ex org".
"ex ng2": Added an activity diagram to the Global function of "test.c" in "ex org".
"ex ok": move LOCAL_MACRO from "test.c" to "test.h" in "ex org"

[example2.zip](https://github.com/doxygen/doxygen/files/12878216/example2.zip)

**Comment by albert-github** [comment]
How is this related to plantuml (and typedef) as the only lines where I see plantuml mentioned is in the `Doxyfile`s  with the line:
```
PLANTUML_JAR_PATH      = $(PLANTUML_INSTALL_DIR)/plantuml.jar
```

furthermore the examples "org", "ng1" and "ng2" only differ is some empty lines and they all differ from the example "ok" as they have the definition of the `LOCAL_MACRO` in them.

The fact that you didn't see any response in #10305 was due to the above reason as the example2 was seen as another example of the same issue.

When the issue is concerning plantuml and typedef is should not have been made part of the issue #10305 but a new issue should have been created geared to the specific problem.

It looks like that the given example here is missing some parts.

**Comment by Yoshi-Nakai** [comment]
Sorry for the late reply.

I have prepared test.Doxyfile separately for the purpose of incorporating PlantUML functionality.
Use the same test.Doxyfile.
We have also prepared a new batch file to run Doxygen.

1. With "ex org", there is no warning in test.Doxy.err and you get the expected documentation.

2. For "ex ng1", a warning is output to test.Doxy.err, and the Macros section of test_8c.html has disappeared.

3. In "ex ng2", there is no warning in test.Doxy.err, but the e_test_function2() description in test_8c.html and the e_test_function2() description in test_8h.html are different.

4. With "ex ok", there will be no warning in test.Doxy.err. Although the expected documentation is obtained, it is undesirable to include the Macros section in test_8h.html.

5. In "ex ng3", LOCAL_MACRO was moved after the activity diagram of "test.h". A warning was output and the Macros section of test_8h.html disappeared.

[example3.zip](https://github.com/doxygen/doxygen/files/12926148/example3.zip)

**Comment by albert-github** [comment]
I used the tests from example3.zip

- I don't see anywhere the word `typedef` in the examples, did you mean `#define`?
- the example parts `ex ng1` and `ex_ng3` have the same background, for this I've just pushed a proposed patch, pull request #10366
- the problem mentioned in `ex ng2` is not related to `plantuml` but to the general way the different comment blocks are joined (this could / should be improved) and is a different issue (might be related to the initial question in #10305).

**Comment by Yoshi-Nakai** [comment]
very sorry.
I intended to write Define, but unconsciously wrote TYPEDEF.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
